cime_config/cam.case_setup.py

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python3
+
+"""Copy GEOS-Chem configuration files from source to the case directory.
+This script is run from CIME when calling case.setup"""
+
+import logging
+import os
+import shutil
+import sys
+
+_CIMEROOT = os.environ.get("CIMEROOT")
+if _CIMEROOT is None:
+    raise SystemExit("ERROR: must set CIMEROOT environment variable")
+# end if
+_LIBDIR = os.path.join(_CIMEROOT, "CIME", "Tools")
+sys.path.append(_LIBDIR)
+sys.path.insert(0, _CIMEROOT)
+
+#pylint: disable=wrong-import-position
+from CIME.case import Case
+
+logger = logging.getLogger(__name__)
+
+if len(sys.argv) != 3:
+    raise SystemExit(f"Incorrect call to {sys.argv[0]}, need CAM root and case root")
+# end if
+cam_root = sys.argv[1]
+case_root = sys.argv[2]
+
+with Case(case_root) as case:
+    cam_config = case.get_value('CAM_CONFIG_OPTS')
+    # Gather case information (from _build_usernl_files in case_setup.py)
+    comp_interface = case.get_value("COMP_INTERFACE")
+
+    if comp_interface == "nuopc":
+        ninst = case.get_value("NINST")
+    elif ninst == 1:
+        ninst = case.get_value("NINST_CAM")
```

Comment:
Could you explain what ninst is and how it is different for nuopc versus non-nuopc?

Reply:
CESM has the capability to run an ensemble of instances as a single run. This is most often used as part of a data assimilation (DA) cycle. The ensemble runs, say for 6 hours, then writes restart files and stops while [DART](https://dart.ucar.edu/) updates the state. Next, the model restarts with the updated state. This entire process can run several times as a single job submission (longer runs are handled with automatic job resubmission).

When the multiple instance implementation was first created in MCT, one idea was to be able to have some components run a single instance while others ran several (for instance, a single ocean model responding to an ensemble of atmosphere simulations). However, no one ever figured out how to define this scientifically (coupling is hard) so the idea was officially dropped many years ago. The move to NUOPC just formalizes that situation.

The impact on GEOS-CHEM is deciding if it will support different configurations for different instances of CAM running as a single job. CAM will have `atm_in_0001`, `atm_in_0002` etc. but that has an impact inside of CAM where it has to check its status as an instance and look for the correct namelist filename to open. I have no idea if this is a priority for the community that will be using GEOS-CHEM which is why they should decide if the (not insignificant) extra effort is worth it. If they decide they do not need full multi-instance support, I suggest it is either prevented (perhaps in this script) or it is just specified that all instances will use the same set of GEOS-CHEM configuration files.

Reply:
Not quite. I think you need to remove lines 56-59 (and 61).
Perhaps leave a comment saying that those files will be the same on all instances for a multi-instance case.